schematics/ng-componize/htmlParser.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { expect } from 'chai';
+import {
+  ConfigurationName,
+  extractParameterName,
+  launchParsing,
+  parserFunctionsConfigurations,
+} from './HtmlParser';
```

Comment:
Thanks a lot for your contribution.
So, here you are importing the wrong file, H letter should be lowercase instead of uppercase [htmlParser](https://github.com/bilelmsekni/componizer/blob/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54/schematics/ng-componize/htmlParser.ts).

Reply:
done, thanks for review

Reply:
You're welcome :smile:.
So @hugoparis19 I had a better time to review your pull request now. I am so glad with that new feature, it looks very good, but I found some issues and I would like your fix them.

1. We need correct import here [htmlParser](https://github.com/bilelmsekni/componizer/pull/12/commits/71c0a5025b1015f06de19f23ef58e0b5e6be8e9e#diff-b84399c1f453ef11cc57bf7c3be53087d313907161bd05d991a3673fcbb147ebL14) too. Sorry for I not seen it before.
2. [findNewComponentPrefix](https://github.com/bilelmsekni/componizer/pull/12/commits/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54#diff-b84399c1f453ef11cc57bf7c3be53087d313907161bd05d991a3673fcbb147ebR24) is not getting component prefix well. I think you should use the **WorkspaceSchema** intead of **WorkspaceProject**. You can debug or log the [project](https://github.com/bilelmsekni/componizer/pull/12/commits/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54#diff-b84399c1f453ef11cc57bf7c3be53087d313907161bd05d991a3673fcbb147ebR22) variable and [workSpace](https://github.com/bilelmsekni/componizer/pull/12/commits/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54#diff-b84399c1f453ef11cc57bf7c3be53087d313907161bd05d991a3673fcbb147ebR22) variable to see how you can get correct prefix.
3. I was getting a error when testing extension. `yarn ext:test`

```
TypeError: mocha.useColors is not a function
	at Object.run (/home/maykon-oliveira/Projects/componizer/extension/out/test/suite/index.js:11:11)
```
But it's easy to fix. I don't know why `mocha.useColors(true);` is not working for me. Is it working well for you?
```
const mocha = new Mocha({
	ui: 'tdd',
	useColors: true
});
```
4. Fourth and last. I think we could (or couldn't) think better names for interfaces, functions and variables. I have some ideas, but I really would like your help too.

interface **[ConfigurationName](https://github.com/bilelmsekni/componizer/pull/12/commits/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54#diff-393ee30072ed287eaec8f839b66d3d51266bc0492781e86a3a31a35144f393aeR15)**
- BindingType?
- BindingTypeLookup?

interface **[RegexpConfiguration](https://github.com/bilelmsekni/componizer/pull/12/commits/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54#diff-393ee30072ed287eaec8f839b66d3d51266bc0492781e86a3a31a35144f393aeR105)**
- BindingTypeRegexpConfiguration?
- BindingTypeRegexp?

interface **[FoundItem](https://github.com/bilelmsekni/componizer/pull/12/commits/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54#diff-393ee30072ed287eaec8f839b66d3d51266bc0492781e86a3a31a35144f393aeR4)**
- BindingTypeMatchItem?
- BindingMatchConfiguration?

Please let me know if it's not making sense for you.

Reply:
Thanks for your comments, I am totally agree with you, unfortunately I will be able to make the changes on Tuesday only (I'm on holiday)

Reply:
I can confirm that the moch.useColors is not working for me too, using @maykon-oliveira 's solution fixed the issue

Reply:
I managed to push the fix to the branch

Reply:
@maykon-oliveira I figured out a way to fix your #7 . By changing the peerDependencies of the schematics to dependencies, now we are sure to always use the right version of @schematics/angular even if the project is on 10.x.y

Reply:
closes #7

Reply:
@maykon-oliveira I logged project variable and workSpace variable to find the correct prefix ([findNewComponentPrefix](https://github.com/bilelmsekni/componizer/pull/12/commits/1d0dc18cce0c33dcb2f921d780a7e20421c2ec54#diff-b84399c1f453ef11cc57bf7c3be53087d313907161bd05d991a3673fcbb147ebR24)) but the project exposes the correct prefix value on my examples. What is your case where the prefix isn't correct ?

Reply:
@bilelmsekni @maykon-oliveira , I just push with @ekaliroots the renaming of the various interfaces and the split of the htmlParser :

ConfigurationName => BindingTypeLookup

RegexpConfiguration => BindingTypeLookupRegexp

FoundItem => BindingMatch

We only have the comment about the prefix open, but I let @maykon-oliveira answer to my previous question.

Reply:
The tests are :heavy_check_mark: .

I made a push to resolve some things.

- Fixes htmlParser wrong import
- Removes unnecessary whitespace between open tag name and tag attributes.
- New unit test to verify if schematic replace selected content to the new tag of created component. This one will confirm that the prefix problem I was facing not exist.

I'd like to create others tests too, the more, the merrier.